eng/native/configurecompiler.cmake

```diff
@@ -1027,3 +1034,8 @@ elseif (NOT CLR_CMAKE_HOST_BROWSER AND NOT CLR_CMAKE_HOST_WASI)
     enable_language(ASM)
 
 endif(CLR_CMAKE_HOST_WIN32)
+
+if (INTERNAL_ZLIB_NG)
+  add_definitions(-DINTERNAL_ZLIB_NG)
```

Comment:
Do we use this `INTERNAL_ZLIB_NG` define anywhere?

Reply:
IIRC I added this when we were starting to discuss the mobile split, and I was bringing back the preprocessor directive conditions in C files. Then I realized those conditions in C code were not needed anymore.

I also have similar lines in mono (src/mono/CMakeLists.txt) and in nativeaot (src/native/libs/System.IO.Compression.Native/CMakeLists.txt).

Maybe none of those 3 are needed anymore. Let me remove them in a commit and see if the CI fails again.

The ZLIB_COMPAT line is needed though, that one is consumed by the zlib-ng code itself.

Reply:
Actually, it seems both can go away. ZLIB_COMPAT is already being set in zlib-ng.cmake. I'll also test that removal.